### PR TITLE
[bugfix] fix idempotency for mso_schema_site_bd_l3out

### LINKED ISSUE #347: Module `mso_schema_site_bd_l3out` is not idempotent
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

* The `mso_schema_site_bd_l3out` module doesn't seem to be idempotent. It applies duplicate l3out's to the bridge domain.

### Affected Module Name(s):

* mso_schema_site_bd_l3out

### MSO version and MSO Platform

* V 2.1(2d) and all.

### APIC version and APIC Platform for Site Level Resources

* V x.x.x and on-prem/cloud-aws/cloud-azure/all.

### Collection versions

* cisco.mso 2.2.1

### Output/ Error message

```bash
changed: [lab_host] => {
    "changed": true,
    "current": {
        "l3outName": "LAB_L3OUT",
        "schemaId": "603e68151300003b5c7e835e",
        "templateName": "LAB_TEMPLATE"
    },
    "invocation": {
        "module_args": {
            "bd": "LAB_BD",
            "host": "lab_host",
            "hostname": "lab_host",
            "l3out": {
                "name": "LAB_L3OUT",
                "schema": "LAB_SCHEMA",
                "template": "LAB_TEMPLATE"
            },
            "login_domain": null,
            "output_level": "normal",
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": null,
            "schema": "LAB_SCHEMA",
            "site": "LAB_SITE",
            "state": "present",
            "template": "LAB_TEMPLATE",
            "timeout": 30,
            "use_proxy": null,
            "use_ssl": null,
            "username": "LAB_USERNAME",
            "validate_certs": false
        }
    }
}
```

### Expected Behavior

* I would expect there not to be a change as the l3out was already applied to the bridge domain

### Actual Behavior

* The same l3out got applied to the bridge domain. Every time the module is run it adds another one.

### Playbook tasks to Reproduce

*
```yaml
- name: MSO BRIDGE DOMAIN TO L3OUT
  cisco.mso.mso_schema_site_bd_l3out:
    hostname: "{{ inventory_hostname }}"
    username: "{{ ansible_user }}"
    password: "{{ ansible_ssh_pass }}"
    validate_certs: "{{ validate_certs }}"
    schema: "{{ schema }}"
    template: "{{ template }}"
    site: "{{ site }}"
    bd: "{{ bd_name }}"
    l3out:
        name: "{{ l3out_name }}"
        schema: "{{ schema }}"
        template: "{{ template }}"
    state: "present"

- name: MSO BRIDGE DOMAIN TO L3OUT
  cisco.mso.mso_schema_site_bd_l3out:
    hostname: "{{ inventory_hostname }}"
    username: "{{ ansible_user }}"
    password: "{{ ansible_ssh_pass }}"
    validate_certs: "{{ validate_certs }}"
    schema: "{{ schema }}"
    template: "{{ template }}"
    site: "{{ site }}"
    bd: "{{ bd_name }}"
    l3out:
        name: "{{ l3out_name }}"
        schema: "{{ schema }}"
        template: "{{ template }}"
    state: "present"
```

### Important Factoids

You can see that in the output it has the **schemaId** instead of just schema, which leads me to believe it is converting the schema to an ID and using that for comparison. When I put in the ID in for `schema` it gave me the following error:

```
"Provided schema '603e68151300003b5c7e835e' does not exist."
```

### References

* #0000

### PR DESCRIPTION
**Pull request by akinross**
fixes #347

### CODE REVIEW

File: tests/integration/targets/mso_schema_site_bd_l3out/tasks/main.yml [hunk 1]
```diff
@@ -58,6 +58,7 @@
     - '{{ mso_site | default("ansible_test") }}'
     - 'aws_{{ mso_site | default("ansible_test") }}'
     - 'azure_{{ mso_site | default("ansible_test") }}'
+  ignore_errors: yes
```
Comment (anvitha-jain) [inline tests/integration/targets/mso_schema_site_bd_l3out/tasks/main.yml:61]: Technically, this case shouldn't fail right?

**Review by anvitha-jain — CHANGES_REQUESTED** [review]

File: tests/integration/targets/mso_schema_site_bd_l3out/tasks/main.yml (hunk 1 above)
Comment (akinross) [reply tests/integration/targets/mso_schema_site_bd_l3out/tasks/main.yml]: have removed was due to azure site not being reachable and thus failing

**Review by anvitha-jain — DISMISSED** [review]
LGTM

File: plugins/modules/mso_schema_site_bd_l3out.py [hunk 2]
```diff
@@ -239,14 +237,14 @@ def main():
...
+            mso.existing = mso.make_reference(l3out, "l3out", l3out_schema_id, l3out_template)
+        elif not mso.existing:
             l3outs.append(l3out.get("name"))
             l3out_refs.append(l3out_ref)
             ops.append(dict(op="replace", path="{0}/{1}".format(bd_path, bd), value=mso_objects.get("site_bd").details))
-            mso.existing = mso.make_reference(l3out, "l3out", mso_schema.id, template)
-        elif not mso.existing:
-            ops.append(dict(op="add", path="{0}/-".format(bd_path), value=payload))
-            mso.existing = mso.make_reference(l3out, "l3out", mso_schema.id, template)
+            mso.existing = mso.make_reference(l3out, "l3out", l3out_schema_id, l3out_template)
```
Comment (shrsr) [inline plugins/modules/mso_schema_site_bd_l3out.py:247]: Can we specify this-> mso.existing = mso.make_reference(l3out, "l3out", l3out_schema_id, l3out_template)
only once?

**Review by lhercot — APPROVED** [review]
Approving conditional on Shreyas comment

File: plugins/modules/mso_schema_site_bd_l3out.py (hunk 2 above)
Comment (akinross) [reply plugins/modules/mso_schema_site_bd_l3out.py]: agree have changed

**Review by shrsr — APPROVED** [review]
LGTM

**Review by anvitha-jain — APPROVED** [review]

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**